rs/compression/src/noc/noc.rs

```diff
@@ -0,0 +1,55 @@
+use std::fs::File;
+use std::io::{BufWriter, Write};
+
+use anyhow::{anyhow, Result};
+use utils::io::wrap_write;
+
+use crate::compression::Compressor;
+
+pub struct NoCompressor {
+    size: usize,
+    sequence: Vec<u64>,
```

Comment:
is it possible to have the nocompressor to not allocating any data? just have a ref here

Reply:
Actually this is a lot harder than it seems, with all the lifetime management parameters and stuff. And the data will only live temporarily anyway (we create the encoder to encode and write down the encoded data to disk all in one place), maybe it doesn't really matter? We proceed one posting list at a time, so it shouldn't take too much memory. As for runtime, yeah we still have to pay for the copy. But since a posting list shouldn't be too big, I hope it's not too big of a deal.
Maybe you have an idea how to make NoEncoder not own the data? But conceptually, it would make more sense for the encoder to own the encoded data, right?